ci/audit-logs-s3.py

```diff
@@ -0,0 +1,126 @@
+    )
+    pages=json.loads(cf_json)
+    total_results = pages['pagination']["total_results"]
+    if total_results > 0:
+        total_pages= pages['pagination']["total_pages"]
+        for page in range(total_pages):
+            result= subprocess.check_output(
+            "cf curl '/v3/audit_events?created_ats[gt]=" + str(start) + "&created_ats[lt]=" + str(end) + "&order_by=created_at&page=" + str(page+1)+"'",
+            universal_newlines=True,
+            shell=True,
+            )
+            data= json.loads(result)
+            result_data=data.get("resources",{})
+            audit_logs.extend(result_data)
+        return audit_logs
+    else:
+        return "empty"
+
+@functools.cache
+def get_cf_entity_name(entity, guid):
+    """
+    Retrieves the name of a CF entity from a GUID.
+    """
+    if not guid:
+        return
+    elif 'guid' in guid:
+        guid = ast.literal_eval(guid)['guid']
+        cf_json = subprocess.check_output(
+            "cf curl /v3/" + entity + "/" + guid,
+            universal_newlines=True,
+            shell=True,
+        )
+        cf_data = json.loads(cf_json)
+        return cf_data.get("name", "N/A")
+    else:
+        return "None"
+
+
+def upload_to_s3(bucket_name, object_name, data):
+    try:
+        body = '\n'.join([
+                    json.dumps({
+                           **{k: v for k,v in item.items() if k not in ["links"]},
+                           "organization_name": get_cf_entity_name("organizations", f"{item['organization']}"),
+                           "space_name": get_cf_entity_name("spaces", f"{item['space']}")
+                           })
+                          for item in data
+                          ])
+        s3_client.put_object(
+            Bucket=bucket_name,
+            Key=object_name,
+            Body=body,
+            ContentType='application/json'
+            )
+        print("success for time "+ str(start_time))
+    except Exception as e:
+        print(traceback.format_exc())
+        print(f'Error upload file to S3 for time starting' + str(start_time) + " and end time" + str(end_time))
+        exit(1)
+
+def update_latest_stamp_in_s3(latest_timestamp):
+    data = latest_timestamp
+    s3_client.put_object(
+            Bucket=bucket_name,
+            Key=timestamp_key,
+            Body=data,
+            )
+
+def main():
+    try:
+        audit_logs = get_audit_logs(start_time,end_time)
+        if audit_logs != "empty":
+            timestamp=audit_logs[-1]['created_at']
+            object_name = f"{now.year}/{now.month:02d}/{now.day:02d}/{now.hour:02d}/{now.minute:02d}/{now.second:02d}"
+            upload_to_s3(bucket_name, object_name,audit_logs)
+            update_latest_stamp_in_s3(timestamp)
+        else:
+            print("empty list")
+            update_latest_stamp_in_s3(end_time)
+        # update latest timestamp on success
+    except Exception as e:
+        print("error " + str(e))
+        exit(1)
```

Comment:
I don't think we need a `try/except` block here. If any errors are raised by the code, then the program should write the error to stderr and exit 1